portfolio/src/main/webapp/script.js

```diff
@@ -82,3 +73,43 @@ function createListElement(text) {
   liElement.innerText = text;
   return liElement;
 }
+
+
+function getComment() {
+  var limit = document.getElementById("num-results").value;
+  var limitParsed = +limit;
+   
+  fetch('/list-post').then(response => response.json()).then((comment) => {
+     const statsListElement = document.getElementById('post-list');
+     statsListElement.innerHTML = '';
+
+     //Get the right value of the limit based on the user's input.  
+     var difference;
+    if(comment.length < limitParsed || comment.length == limitParsed){
+    difference = comment.length;
+    }else{
+    difference = limitParsed;
+    }
+
+    //Output every comment until reaching the limit
+    var i;
+    for (i = 0; i < difference; i++) {
```

Comment:
you should be able to put the initialization of "i" directly in the for loop

Reply:
Alright.